flake.nix

```diff
@@ -0,0 +1,108 @@
+{
+  description = "slugger: Clean URI slugs for Haskell";
+
+  nixConfig.bash-prompt = "[nix]λ ";
+
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixos-21.05";
+
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { flake-utils, nixpkgs, self }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        config = {};
+
+        compilerVersion = "884";
+        compiler = "ghc" + compilerVersion;
+        overlays = [
+          (final: prev: {
```

Comment:
Just a convention; these tend to be called `self: super:`.

Reply:
The latest nix recommendations that I've asked about and seen in docs tend to use `final` and `prev` now...

https://nixos.wiki/wiki/Flakes (no mention of super)

https://nixos.wiki/wiki/Overlays (has both)

Reply:
Interesting, I've never seen it before. I guess I am just old :).

Nixpkgs seems to be mostly using `self: super:`, but `final: prev:` is showing up too:

 ```
$ cd nixpkgs/ 
$ rg 'self: super:' | wc -l                                                                                                                                                   
223
$ rg 'final: prev:' | wc -l                                                                                                                                                   
11
```